wrangler-api/src/main/java/io/cdap/wrangler/api/schema/SchemaUpdater.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler.api.schema;
+
+import io.cdap.cdap.api.data.schema.Schema;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This interface represents an entity that changes the schema of input data after applying a transformation.
+ */
+public interface SchemaUpdater {
+  Map<String, Schema> ADDED_FIELDS = new HashMap<>();
```

Comment:
You can't define fields in the interface. This becomes global singletons and I think this is not what you want

Reply:
Yeah, was just experimenting, thought I'll keep a new `AbstractSchemaUpdater` class that the specific directives can implement. 

I think the easiest approach is to just have an interface with four methods to get collections of added, removed, renamed and type changed fields. They return empty by default, which is backwards compatible as we will generate schema in that case. If the UDD wants to implement the method, they have to override and return custom maps/lists.